draft-ietf-masque-h3-datagram.md

```diff
@@ -303,270 +202,84 @@ Capsule Value:
 : The payload of this capsule. Its semantics are determined by the value of the
 Capsule Type field.
 
-
-## Requirements
-
-If the definition of an HTTP Method or HTTP Upgrade Token states that it uses
-the capsule protocol, its implementations MUST follow the following
-requirements:
-
-* A server MUST NOT send any Transfer-Encoding or Content-Length header fields
-  in a 2xx (Successful) response. If a client receives a Content-Length or
-  Transfer-Encoding header fields in a successful response, it MUST treat that
-  response as malformed.
-
-* A request message does not have content.
-
-* A successful response message does not have content.
-
-* Responses are not cacheable.
-
-## Intermediary Processing
-
-Intermediaries MUST operate in one of the two following modes:
-
-Pass-through mode:
-
-: In this mode, the intermediary forwards the data stream between two
-associated streams without any modification of the data stream.
-
-Participant mode:
-
-: In this mode, the intermediary terminates the data stream and parses all
-Capsule Type and Capsule Length fields it receives.
-
-Each Capsule Type determines whether it is opaque or transparent to
-intermediaries in participant mode: opaque capsules are forwarded unmodified
-while transparent ones can be parsed, added, or removed by intermediaries.
-Intermediaries MAY modify the contents of the Capsule Data field of transparent
-capsule types.
-
-Unless otherwise specified, all Capsule Types are defined as opaque to
-intermediaries. Intermediaries MUST forward all received opaque CAPSULE frames
-in their unmodified entirety. Intermediaries MUST NOT send any opaque CAPSULE
-frames other than the ones it is forwarding. All Capsule Types defined in this
-document are opaque, with the exception of the datagram capsules, see
-{{datagram-capsule}}. Definitions of new Capsule Types MAY specify that the
-newly introduced type is transparent. Intermediaries MUST treat unknown Capsule
-Types as opaque.
-
-Intermediaries respect the order of opaque CAPSULE frames: if an intermediary
-receives two opaque CAPSULE frames in a given order, it MUST forward them in
-the same order.
+Capsules MUST be forwarded unmodified by intermediaries, with the exception of
+the DATAGRAM capsule; see {{datagram-capsule}}. An intermediary can identify the
+use of the capsule protocol either through the presence of the Capsule-Protocol
+header field ({{hdr}}) or by understanding the chosen HTTP Upgrade token. An
+intermediary that identifies the use of the capsule protocol MAY convert between
+DATAGRAM capsules and QUIC DATAGRAM frames when forwarding. Definitions of new
+Capsule Types MAY specify optional custom intermediary processing.
```

Comment:
I don't recall us having consensus on this text around intermediaries. Can you please point me to the minutes or email thread where we agreed on this?

Reply:
We reached rough consensus on this at the second design team meeting on 2021-11-29: https://github.com/ietf-wg-masque/draft-ietf-masque-h3-datagram/pull/124#discussion_r758842704

Reply:
This is rather less than minutes. I'll send mail.

Reply:
Discussed over email, and resolved by making the intermediary requirement a SHOULD with explanations